geoutils/satimg.py

```diff
@@ -283,56 +306,61 @@ def __init__(self, filename_or_dataset, attrs=None, load_data=True, bands=None,
 
         self.__get_date()
 
-    def __get_date(self):
+    def __get_date(self) -> dt.datetime | None:  # type: ignore
 
         """
         Get date from datetime
         :return:
         """
         if self.datetime is not None:
-            self.date = self.datetime.date()
+            self.date = self.datetime.date()  # type: ignore
         else:
             self.date = None
 
-    def __parse_metadata_from_fn(self,silent=False):
+    def __parse_metadata_from_fn(self, silent: bool = False) -> None:
 
         """
         Attempts to pull metadata (e.g., sensor, date information) from fname, setting sensor, satellite,
         tile, datetime, and date attributes.
         """
-
         fname = self.filename
-        name_attrs = ['satellite', 'sensor', 'product', 'version', 'tile_name', 'datetime']
-        attrs = parse_metadata_from_fn(fname)
+        name_attrs = ["satellite", "sensor", "product", "version", "tile_name", "datetime"]
+        attrs = parse_metadata_from_fn(fname if fname is not None else "")
```

Comment:
wow, was this "if loop" made automatically for the cases with `fname=None`? :O 
would be impressive!

Reply:
No that was me haha. `parse_metadata_from_fn` raises a `ValueError` if None is given. So I made that fix to pass an empty string instead.